pose_evaluation/metrics/test_embedding_distance_metric.py

```diff
@@ -0,0 +1,409 @@
+
+
+def test_mixed_input(cosine_metric):
+    """Test score function with mixed input types."""
+    emb1 = np.random.rand(768)
+    emb2 = torch.rand(768)
+
+    score = cosine_metric.score(emb1, emb2)
+    assert isinstance(score, float), "Output should be a float."
+
+
+@pytest.mark.parametrize("num_points, dim", [(16, 2)])
+def test_unit_circle_points(cosine_metric, num_points, dim):
+    embeddings = generate_unit_circle_points(num_points, dim)
+    distances = cosine_metric.score_all(embeddings, embeddings)
+    save_and_plot_distances(distances=distances, matrix_name="Unit Circle", num_points=num_points, dim=dim)
+
+
+@pytest.mark.parametrize("num_points, dim", [(20, 2)])
+def test_orthogonal_rows_with_repeats_2d(cosine_metric, num_points, dim):
+    embeddings = generate_orthogonal_rows_with_repeats(num_points, dim)
+    distances = cosine_metric.score_all(embeddings, embeddings)
+    save_and_plot_distances(
+        distances=distances, matrix_name="Orthogonal Rows (with repeats)", num_points=num_points, dim=dim
+    )
+
+    # Create expected pattern directly within the test function
+    expected_pattern = torch.zeros(num_points, num_points, dtype=distances.dtype)
+    for i in range(num_points):
+        for j in range(num_points):
+            if (i + j) % 2 != 0:
+                expected_pattern[i, j] = 1
+
+    # We expect 0 1 0  across and down
+    assert torch.allclose(
+        distances, expected_pattern, atol=1e-6
+    ), "Output does not match the expected alternating pattern"
+
+
+@pytest.mark.parametrize("num_points, dim", [(20, 2)])
+def test_orthogonal_rows_in_pairs(cosine_metric, num_points, dim, distance_range_checker):
+    embeddings = generate_orthogonal_rows_in_pairs(num_points, dim)
+    distances = cosine_metric.score_all(embeddings, embeddings)
+    save_and_plot_distances(distances, "orthogonal_rows_in_pairs", num_points, dim)
+    distance_range_checker(distances, min_val=0, max_val=2)  # Check distance range
+
+
+@pytest.mark.parametrize("num_points, dim", [(10, 5)])
+def test_ones_tensor(cosine_metric, num_points, dim, distance_range_checker):
+    embeddings = generate_ones_tensor(num_points, dim)
+    distances = cosine_metric.score_all(embeddings, embeddings)
+    save_and_plot_distances(distances, "ones_tensor", num_points, dim)
+    distance_range_checker(distances, min_val=0, max_val=0)  # Expect all distances to be 0
+
+
+@pytest.mark.parametrize("num_points, dim", [(15, 15)])  # dim should be equal to num_points for identity matrix
+def test_identity_matrix_rows(cosine_metric, num_points, dim, distance_range_checker):
+    embeddings = generate_identity_matrix_rows(num_points, dim)
+    distances = cosine_metric.score_all(embeddings, embeddings)
+    save_and_plot_distances(distances, "identity_matrix_rows", num_points, dim)
+    distance_range_checker(distances, min_val=0, max_val=2)  # Check distance range
+
+
+# def test_progress_bar(cosine_metric):
```

Comment:
why commented?

Reply:
Well, the progress bar currently does nothing, but if we ever implement it we'll want to test that